tools/run-conftest.sh

```diff
@@ -16,36 +16,25 @@ while [[ $# -gt 0 ]]; do
     shift # skip --policies-path
     shift # skip policies-path value
     ;;
-  --temp-dir)
-    TEMP_DIR="$2"
-    shift # skip --temp-dir
-    shift # skip temp-dir value
-    ;;
-  --do-dependency-update)
-    DO_DEPENDENCY_UPDATE=1
-    shift # skip --do-dependency-update
-    ;;
   *)
     break
     ;;
   esac
 done
 
-CHART_PATH=$GIT_REPO_ROOT/operations/helm/charts/mimir-distributed
-VALUES_FILES_PATH=$GIT_REPO_ROOT/operations/helm/charts/mimir-distributed/ci
-
-if [[ -z "$TEMP_DIR" ]]; then
-  TEMP_DIR=$(mktemp -d)
+if ! [ -d "$GENERATED_DIR" ] ; then
+  echo "The generated yaml templates in $GENERATED_DIR are not present, use 'make check-helm-tests' instead of calling this script directly"
+  exit 1
 fi
 
-if (( $DO_DEPENDENCY_UPDATE )); then
-  helm dependency update $CHART_PATH >/dev/null
-fi
+CHART_PATH=$GIT_REPO_ROOT/operations/helm/charts/mimir-distributed
+VALUES_FILES_PATH=$GIT_REPO_ROOT/operations/helm/charts/mimir-distributed/ci
 
-for file in $(find $VALUES_FILES_PATH -name '*.yaml'); do
-  echo -n "Testing with values file $file"
-  output_dir=$TEMP_DIR/$(basename "$file")/
-  helm template test $CHART_PATH -f $file --output-dir $output_dir --namespace citestns >/dev/null
-  conftest test $output_dir -p $POLICIES_PATH --combine --ignore '.*/mimir-distributed/charts/.*'
+for FILEPATH in $(find $VALUES_FILES_PATH -name '*.yaml'); do
+  TEST_NAME=$(basename -s '.yaml' "$FILEPATH")
+  OUTPUT_DIR="${GENERATED_DIR}/${TEST_NAME}-generated"
```

Comment:
This is also implicit in the two scripts. I wonder if we even need this.. I think just discovering directories in operations/helm/tests and outputting the basename of each directory is sufficient to let the person know which test failed (this may also require moving the intermediate directory somewhere outside of operations/helm/tests)

Reply:
also output dir doesn't make much sense now. Previously it was used for helm to output there, but now it's actually the input dir for run-conftest.sh